src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransPatterns.java

```diff
@@ -534,19 +534,6 @@ private void handleSwitch(JCTree tree,
                         JCExpression test = (JCExpression) this.<JCTree>translate(label.pat);
                         if (label.guard != null) {
                             JCExpression guard = translate(label.guard);
-                            if (hasJoinedNull) {
-                                JCPattern pattern = label.pat;
-                                while (pattern instanceof JCParenthesizedPattern parenthesized) {
```

Comment:
side: this idiom seems to be useful, currently used in a couple of places in Check, could it make sense to extract it into a method?